go/pools/smartconnpool/stack.go

```diff
@@ -54,6 +54,11 @@ func (s *connStack[C]) Pop() (*Pooled[C], bool) {
 	}
 }
 
+func (s *connStack[C]) Peek() *Pooled[C] {
```

Comment:
Some comments on the implementation in this file will be nice. I don't fully understand popCount just from reading the file contents.